### ISSUE TITLE
Renaming the init module

### ISSUE THREAD

**Issue by markf94** [XanaduAI/pennylane#574]
As I'm working through the PennyLane tutorial pages I encountered the `pennylane.init` module. I encountered it for the first time when used like this:

```
from pennylane.init import strong_ent_layers_uniform
```

This might be either my inner (tidy) German or my strong feelings about clean & readable Python code but there is something wrong with this. It just doesn't fit into the otherwise very clean and beautiful `pennylane` package. Therefore, I wanted to ask how you would feel about renaming this module to `parameter_initialization` so that the import statement looks like:

```
from pennylane.parameter_initialization import strong_ent_layers_uniform
```

I'm more than happy to unleash my need for tidiness and implement this change if you are okay with it.

**Comment by josh146** [comment]
Thanks @markf94! This is something that is on our roadmap to think about and clarify --- this includes moving the initialization functions 'closer' to the actual templates (options include even providing the initialization functions as methods on the templates themselves).

**Comment by markf94** [comment]
Make sense! So I shouldn't rename this module right now?

**Comment by josh146** [comment]
Renaming the module here would also require changes made to the tutorials over at [XanaduAI/qml](https://github.com/XanaduAI/qml), so it might make sense to first discuss and decide if we are going to make more drastic changes to the structure/UI of the initialization functions.

Tagging @mariaschuld to also get her thoughts!

**Comment by mariaschuld** [comment]
As mentioned in Issue #575, we are planning to overhaul this module soon.

But since this came up a couple of times (for example shortening the function names), we could indeed think of an interim solution? My concern about a rename is that it is feature breaking, and we would then introduce another feature break soon with the larger changes...

**Comment by markf94** [comment]
Yeah that's a good point actually. We shouldn't be breaking features too often - I guess it all depends on how soon that big refactor is to be expected. Will this take weeks or months? ;)

**Comment by mariaschuld** [comment]
Months, rather...

I personally would prefer keeping the short name for now (also because it is not likely a severe hindrance for people to use the module), but it is an issue we should keep open to inform the refactor design. 

In any case, thanks @markf94, your input is appreciated!

**Comment by co9olguy** [comment]
I believe the name was chosen to echo similar naming schemes from ML libraries (e.g., the `torch.nn.init` module. 

Tensorflow has `tf.initializers` (deprecated), `tf.keras.initializers`, of just makes things available at top level (`tf.constant_initializer`). 

I think `parameter_initialization` is probably unnecessarily long, but we can look for a balance here if we think it's unclear to a large set of users
